### PR TITLE
README - Update build instructions

### PR DESCRIPTION
**Pull request by jomey**
Tried to build timewarrior from source for the first time using the instructions in the README. I missed seeing the `--recurse-submodules` for the clone option and suggesting an updated instruction line to better highlight that.

### CODE REVIEW

**Comment by lauft** [comment]
Hm... according to the [Git documentation](https://git-scm.com/docs/git-clone#Documentation/git-clone.txt---recurse-submodulesltpathspecgt) of the `git clone` command, the option `--recurse-submodules` should exactly do what the additional `git submodule update --init --recursive` does.

Just to be on the safe side: Are you using Git >1.9? (before the option was `--recursive`) 🤔 

I tested the original command on a Ubuntu Docker container and it works out of the box:
```shell
$ git clone --recurse-submodules https://github.com/GothenburgBitFactory/timewarrior
Cloning into 'timewarrior'...
remote: Enumerating objects: 18097, done.
remote: Counting objects: 100% (2102/2102), done.
remote: Compressing objects: 100% (559/559), done.
remote: Total 18097 (delta 1593), reused 1946 (delta 1522), pack-reused 15995
Receiving objects: 100% (18097/18097), 4.44 MiB | 12.84 MiB/s, done.
Resolving deltas: 100% (13992/13992), done.
Submodule 'src/libshared' (https://github.com/GothenburgBitFactory/libshared.git) registered for path 'src/libshared'
Cloning into '/tmp/timewarrior/src/libshared'...
remote: Enumerating objects: 5225, done.
remote: Counting objects: 100% (1002/1002), done.
remote: Compressing objects: 100% (329/329), done.
remote: Total 5225 (delta 756), reused 848 (delta 652), pack-reused 4223
Receiving objects: 100% (5225/5225), 966.01 KiB | 7.61 MiB/s, done.
Resolving deltas: 100% (4014/4014), done.
Submodule path 'src/libshared': checked out '35e6ec09595c80bc744a27eda50e4dd34d6c7303'
```

**Comment by jomey** [comment]
Ah, now I see my missing piece. When I initially read the instructions, I did not copy the command and overzealously went to my usual 'copy the git URL via the green button'. 

I does work for me with the added option `--recurse-submodules`

Maybe we can keep the updated instruction description line to indicate that there are submodules?

File: README.md [hunk 1]
```diff
@@ -46,10 +46,11 @@ Building Timewarrior yourself requires
 
 There are two ways to retrieve the Timewarrior sources:
 
-* Clone the repository from GitHub,
+* Clone the repository from GitHub and update required submodules,
   ```
   git clone --recurse-submodules https://github.com/GothenburgBitFactory/timewarrior
   cd timewarrior
+  git submodule update --init --recursive
```
Comment (lauft) [inline README.md]: I would remove this unnecessary command.

File: README.md [hunk 2]
```diff
@@ -46,10 +46,11 @@ Building Timewarrior yourself requires
 
 There are two ways to retrieve the Timewarrior sources:
 
-* Clone the repository from GitHub,
+* Clone the repository from GitHub and update required submodules,
```
Comment (lauft) [inline README.md:49]: Ok to this clarification. 👍

**Review by lauft — COMMENTED** [review]
I agree that a bit more clarification will be help, but I would not add unnecessary command to the provided instructions (even if they would not hurt).

File: README.md (hunk 1 above)
Comment (jomey) [reply README.md]: Apologies, I thought my commit got updated with my last push and I had identical modifications for this PR in mind.

**Review by lauft — APPROVED** [review]

**Comment by lauft** [comment]
Thanks for your contribution. ❤️

**Pull request merged**